gpfit/classes.py

```diff
@@ -1,135 +1,420 @@
-"""
```

Comment:
maybe call this module `fits.py` or `fit_classes.py`? Low priority no need to change

Reply:
Yeah I'm not entirely satisfied with this module name either. Tried to address a little with the module docstring, but agree it's not ideal. `fits.py` seems confusingly close to `fit.py`. I like `fit_classes.py` but to me it feels like it still has the same ambiguity of "does it mean python class or class of fit?" and I erred on the side of the succinct. I think I'll sit with `classes.py` for now and if it still feels weird then readdress it in a future release. At least I didn't go full anarchy and call it `function_classes.py` haha.